### ISSUE TITLE
Hevc/x265 freez sometimes in playback

### ISSUE THREAD

**Issue by ramtinak** [Microsoft/FFmpegInterop#76]
Hi, I have change this commit: 02b3be1 but nothings changed..
when I tried to play a Hevc video sometimes freez and then play again.
here is my video info:

Format                         : Matroska
Format version                 : Version 2
File size                      : 242 MiB
Duration                       : 41mn 40s
Overall bit rate               : 811 Kbps
Encoded date                   : UTC 
Writing application            : HandBrake 0.10.2 2015060900
Writing library                : Lavf55.12.0

Video
ID                             : 1
Format                         : HEVC
Format/Info                    : High Efficiency Video Coding
Format profile                 : Main@L3.1
Codec ID                       : V_MPEGH/ISO/HEVC
Width                          : 1 280 pixels
Height                         : 720 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Variable
Original frame rate            : 23.976 fps
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Default                        : Yes
Forced                         : No

Audio
ID                             : 2
Format                         : AAC
Format/Info                    : Advanced Audio Codec
Format profile                 : HE-AAC / LC
Codec ID                       : A_AAC
Duration                       : 41mn 40s
Channel(s)                     : 2 channels
Channel positions              : Front: L R
Sampling rate                  : 48.0 KHz / 24.0 KHz
Compression mode               : Lossy
Delay relative to video        : -105ms
Title                          : Stereo
Default                        : Yes
Forced                         : No

here is the current video link:
http://dl2.film2movie.biz/serial/The%20Flash/S02/The.Flash.S02E16.720p.x265.HEVC.Film2Movie_INFO.mkv

Thanks

**Comment by reego-fr** [comment]
Hi,
does it freeze at a particular timestamp ?

**Comment by ramtinak** [comment]
Hi, thank you for reply
yes, here examples time:(2,6,9,19 sec)

00:00:02 and 00:00:06 and 00:00:09 and 00:00:19

in your MediaPlayerCS is too.
how can I fix this?

**Comment by reego-fr** [comment]
Hi,
I wonder if "Sampling rate : 48.0 KHz / 24.0 KHz" can have something to do with the issue you encountered. 
I don't know what it means but I hope it is not L/R sample rate !

I experienced differently the playback this sample. It freezed, but only after showing people moving at lighting speed, all of them, and with uncompressed audio sample playback it plays well on my side. 
Can you try it again and report about any change ?

**Comment by ramtinak** [comment]
Hi again,
I tested with Uncompressed audio sample too, still not working and still it's freezing

**Comment by reego-fr** [comment]
I find it strange you don't mention anything being better or worse with compressed vs. uncompressed audio samples playback. No offense, but I have to ask you again if you enabled "Force Decode Audio" during your tests. 

I can confirm there is an issue within FFmpegInterop with he-aac audio when playing compressed samples, which is quite easy to fix actually. 

On my side, using a windows 10 desktop, the file plays very fast with bad audio until something gives up a few seconds later and freezes. You might have less processing power than me and experienced it differently. What is your platform and how are x265 480p and 1080p playback on your side ?

**Comment by ramtinak** [comment]
Hi, sorry I forgot to tell you, when I force decode audio (using uncompressed), audio going to fast on blu-ray video. (so getting worse , but freezing still have!). when I disable this option and play this video, audio works well.
Note: I set sample bitrate to 16 so I don't think it's because of this.
----------------------------
I have Windows 10 Pro x64 build 14342 on PC and Windows 10 Mobile build 10.0.14328.1000 on my phone.

On PC after few seconds will freezes but on phone each frame getting freezes and play. so I think there isn't on my side.
--------------------------
HEVC Blu-ray 1080,720 and 480p freezes 
--------------------------
I think the issue is Blu-ray video, I played a WebDL 720p HEVC video, plays very well:
http://onmkv.net/download-alt.php?id=YUhSMGNEb3ZMMVJ5WVdsdVltbDBMbU52YlM5bWFXeGxjeTh6TlRRek1qYzNPRGcwTDBGdVpISnZiaTR5TURFMUxqY3lNSEF1VjBWQ0xVUk1Mbmd5TmpVdVNFVldReTFQVGsxTFZpNXRhM1k=

and I played this video (HDTV 480 HEVC) and plays very well:

General

Format                         : Matroska
Format version                 : Version 4 / Version 2
File size                      : 88.4 MiB
Duration                       : 42mn 30s
Overall bit rate               : 291 Kbps
Movie name                     : PSArips.com | Arrow.S03E22.SD.HDTV.2CH.x265.HEVC-PSA
Encoded date                   : UTC 2015-05-07 04:51:07
Writing application            : mkvmerge v7.8.0 ('River Man') 64bit built on Mar 27 2015 16:31:37
Writing library                : libebml v1.3.1 + libmatroska v1.4.2
DURATION                       : 00:42:25.085000000
NUMBER_OF_FRAMES               : 776
NUMBER_OF_BYTES                : 22819
_STATISTICS_WRITING_APP        : mkvmerge v7.8.0 ('River Man') 64bit built on Mar 27 2015 16:31:37
_STATISTICS_WRITING_DATE_UTC   : 2015-05-07 04:51:07
_STATISTICS_TAGS               : BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES

Video
ID                             : 1
Format                         : HEVC
Format/Info                    : High Efficiency Video Coding
Format profile                 : Main@L3.0
Codec ID                       : V_MPEGH/ISO/HEVC
Width                          : 720 pixels
Height                         : 408 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Variable
Original frame rate            : 23.976 fps
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Default                        : Yes
Forced                         : No
DURATION                       : 00:42:29.714000000
NUMBER_OF_FRAMES               : 61132
NUMBER_OF_BYTES                : 76799368
_STATISTICS_WRITING_APP        : mkvmerge v7.8.0 ('River Man') 64bit built on Mar 27 2015 16:31:37
_STATISTICS_WRITING_DATE_UTC   : 2015-05-07 04:51:07
_STATISTICS_TAGS               : BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES

Audio
ID                             : 2
Format                         : AAC
Format/Info                    : Advanced Audio Codec
Format profile                 : HE-AAC / LC
Codec ID                       : A_AAC
Duration                       : 42mn 30s
Channel(s)                     : 2 channels
Channel positions              : Front: L R
Sampling rate                  : 48.0 KHz / 24.0 KHz
Compression mode               : Lossy
Delay relative to video        : -105ms
Title                          : Stereo
Default                        : Yes
Forced                         : No
DURATION                       : 00:42:30.101000000
NUMBER_OF_FRAMES               : 59768
NUMBER_OF_BYTES                : 15300609
_STATISTICS_WRITING_APP        : mkvmerge v7.8.0 ('River Man') 64bit built on Mar 27 2015 16:31:37
_STATISTICS_WRITING_DATE_UTC   : 2015-05-07 04:51:07
_STATISTICS_TAGS               : BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES

Text
ID                             : 3
Format                         : UTF-8
Codec ID                       : S_TEXT/UTF8
Codec ID/Info                  : UTF-8 Plain Text
Default                        : Yes
Forced                         : No

Here : 
http://up.hidiaries1.ir/download/305328/Arrow.S03E22.SD.HDTV.2CH.x265.HEVC.rar

**Comment by reego-fr** [comment]
Interesting. Does it mean he-aac decoder changed in build 14342 ?
Back to your video issue, would you have a sample of freezing Bluray 480p video ?

**Comment by reego-fr** [comment]
Also, a sample of audio going too fast using uncompressed would be nice as the might be another issue here.

**Comment by reego-fr** [comment]
If we ignore the audio issues with he-aac, this is really a CPU issue. 
It might be possible to enable compressed h265 playback on preview builds though. I am not sure who would be able to confirm this. Maybe @timotiusmargo ?

**Comment by khouzam** [comment]
I can try this and see if h265 compressed is supported.

**Comment by khouzam** [comment]
I'm finding out how to enable the compressed samples support on preview builds. I'm not seeing the freezing issue with the current preview builds. I'm wondering if this is the same as the seek freezing which has been fixed at the pipeline level.

**Comment by reego-fr** [comment]
Do you have details on the availability of the fix you mention ? Are older platforms included as well ?

The freezing and playing issue will occur when buffering is enabled and the cpu can't keep up with decoding samples at a decent rate. 
I didn't try 8k h265 but I am pretty sure there will be an issue with any cpu coming from earth, at least with single threaded decoding.
